src/components/chat/chat_channels.jsx

```diff
@@ -276,6 +270,10 @@ const ChatChannels = React.createClass({
       channels: ChatChannelsStore.getChannels(),
       subscribedChannels: ChatChannelsStore.getSubscribedChannels()
     });
+    // Fetch the initial list of messages for all subscribed rooms.
+    ChatChannelsStore.getSubscribedChannels().map((channel) => {
```

Comment:
@pletcher I just pushed a commit that checks whether a subscribed channel is new, and if so, fetch its list of commits. This is because when the user joins a room, they'll start receiving socket.io notifications for that room. We want the message history to go before the newly incoming messages from the socket. But I agree that this might not be the best place to do this. I'll take a closer look at the problem tomorrow. I'm thinking of just returning the message history from the socket, the moment the user joins the room.